starknet_py/tests/e2e/fixtures/abi_v2_structures.py

```diff
@@ -0,0 +1,193 @@
+    "UserAdded",
+    OrderedDict(user=user_struct),
+)
+
+pool_id_added_dict = {
+    "type": "event",
+    "name": "PoolIdAdded",
+    "inputs": [
+        {"name": "pool_id", "type": "PoolId"},
+    ],
+}
+pool_id_added_event: EventType = EventType(
+    "PoolIdAdded",
+    OrderedDict(pool_id=pool_id_struct),
+)
+
+get_user_dict = {
+    "type": "function",
+    "name": "get_user",
+    "inputs": [
+        {
+            "name": "id",
+            "type": "core::integer::u256",
+        }
+    ],
+    "outputs": [{"type": "User"}],
+}
+get_user_fn = Abi.Function("get_user", OrderedDict(id=UintType(256)), [user_struct])
+
+delete_pool_dict = {
+    "type": "function",
+    "name": "delete_pool",
+    "inputs": [
+        {
+            "name": "id",
+            "type": "PoolId",
+        },
+        {"name": "user_id", "type": "core::integer::u256"},
+    ],
+    "outputs": [],
+}
+delete_pool_fn = Abi.Function(
+    "delete_pool", OrderedDict(id=pool_id_struct, user_id=UintType(256)), []
+)
+
+
+my_struct_dict = {
+    "type": "struct",
+    "name": "test::MyStruct::<core::integer::u256>",
+    "members": [
+        {"name": "a", "type": "core::integer::u256"},
+        {"name": "b", "type": "core::felt252"},
+    ],
+}
+my_struct = StructType(
+    name="test::MyStruct::<core::integer::u256>",
+    types=OrderedDict(a=UintType(256), b=FeltType()),
+)
+
+my_enum_dict = {
+    "type": "enum",
+    "name": "test::MyEnum::<core::integer::u128>",
+    "variants": [
+        {"name": "a", "type": "core::integer::u256"},
+        {"name": "b", "type": "test::MyStruct::<core::integer::u128>"},
+    ],
+}
+my_enum = EnumType(
+    name="test::MyEnum::<core::integer::u128>",
+    variants=OrderedDict(a=UintType(256), b=my_struct),
+)
+
+foo_event_dict = {
+    "type": "event",
+    "name": "foo_event",
+    "inputs": [
+        {"name": "a", "type": "core::felt252"},
+        {"name": "b", "type": "core::integer::u128"},
+    ],
+}
+foo_event = EventType(
+    name="foo_event", types=OrderedDict(a=FeltType(), b=UintType(128))
+)
+
+foo_external_dict = {
+    "type": "function",
+    "name": "foo_external",
+    "inputs": [
+        {"name": "a", "type": "core::felt252"},
+        {"name": "b", "type": "core::integer::u128"},
+    ],
+    "outputs": [{"type": "test::MyStruct::<core::integer::u256>"}],
+    "state_mutability": "external",
+}
+foo_external = Abi.Function(
+    name="foo_external",
+    inputs=OrderedDict(a=FeltType(), b=UintType(128)),
+    outputs=[my_struct],
+)
+
+foo_view_dict = {
+    "type": "function",
+    "name": "foo_view",
+    "inputs": [
+        {"name": "a", "type": "core::felt252"},
+        {"name": "b", "type": "core::integer::u128"},
+    ],
+    "outputs": [{"type": "test::MyEnum::<core::integer::u128>"}],
+    "state_mutability": "view",
+}
+foo_view = Abi.Function(
+    name="foo_view",
+    inputs=OrderedDict(a=FeltType(), b=UintType(128)),
+    outputs=[my_enum],
+)
+
+abi_v2 = Abi(
```

Comment:
If I understand correctly, and we only require `events` for the current tests, then why do we need to define all the other fields?

Reply:
Probably we will need to add more tests for abi v2 so I created the fixture which will be useful in other tests

Reply:
In that case, could you either add more Abi V2 tests in this PR or stick with just the `events` field?